rule-engine/rule-engine-components/src/test/java/org/thingsboard/rule/engine/metadata/TbGetTelemetryNodeTest.java

```diff
@@ -15,72 +15,341 @@
+        config.setFetchMode(fetchMode);
+        config.setLimit(limit);
+
+        node.init(ctxMock, new TbNodeConfiguration(JacksonUtil.valueToTree(config)));
+
+        var actualLimit = ReflectionTestUtils.getField(node, "limit");
+        assertThat(actualLimit).isEqualTo(expectedLimit);
+    }
+
+    private static Stream<Arguments> givenFetchModeAndLimit_whenInit_thenVerifyLimit() {
+        return Stream.of(
+                Arguments.of("FIRST", 0, 1),
+                Arguments.of("LAST", 10, 1),
+                Arguments.of("ALL", 0, 1000),
+                Arguments.of("ALL", 5, 5)
+        );
+    }
+
+    @ParameterizedTest
+    @NullAndEmptySource
+    public void givenEmptyOrderBy_whenInit_thenVerify(String orderBy) throws TbNodeException {
+        config.setOrderBy(orderBy);
+
+        node.init(ctxMock, new TbNodeConfiguration(JacksonUtil.valueToTree(config)));
+
+        var actualOrderBy = ReflectionTestUtils.getField(node, "orderByFetchAll");
+        assertThat(actualOrderBy).isEqualTo("ASC");
+    }
+
+    @Test
+    public void givenConfig_whenInit_thenVerify() throws TbNodeException {
+        List<String> keys = List.of("temperature", "humidity");
+        config.setLatestTsKeyNames(keys);
+        config.setFetchMode("ALL");
+        config.setLimit(5);
+        config.setOrderBy("DESC");
+        config.setAggregation("MIN");
+
+        node.init(ctxMock, new TbNodeConfiguration(JacksonUtil.valueToTree(config)));
+
+        var actualLimit = ReflectionTestUtils.getField(node, "limit");
+        var actualTsKeyNames = ReflectionTestUtils.getField(node, "tsKeyNames");
+        var actualFetchMode = ReflectionTestUtils.getField(node, "fetchMode");
+        var actualOrderByFetchAll = ReflectionTestUtils.getField(node, "orderByFetchAll");
+        var actualAggregation = ReflectionTestUtils.getField(node, "aggregation");
+
+        assertThat(actualLimit).isEqualTo(5);
+        assertThat(actualTsKeyNames).isEqualTo(keys);
+        assertThat(actualFetchMode).isEqualTo("ALL");
+        assertThat(actualOrderByFetchAll).isEqualTo("DESC");
+        assertThat(actualAggregation).isEqualTo(Aggregation.MIN);
+    }
+
+    @Test
+    public void givenEmptyTsKeyNames_whenOnMsg_thenTellFailure() throws TbNodeException {
+        node.init(ctxMock, new TbNodeConfiguration(JacksonUtil.valueToTree(config)));
+        TbMsg msg = TbMsg.newMsg(TbMsgType.POST_TELEMETRY_REQUEST, DEVICE_ID, TbMsgMetaData.EMPTY, TbMsg.EMPTY_JSON_OBJECT);
+        node.onMsg(ctxMock, msg);
+
+        ArgumentCaptor<Throwable> actualException = ArgumentCaptor.forClass(Throwable.class);
+        then(ctxMock).should().tellFailure(eq(msg), actualException.capture());
+        assertThat(actualException.getValue())
+                .isInstanceOf(IllegalStateException.class)
+                .hasMessage("Telemetry is not selected!");
+    }
```

Comment:
We should ensure that time series keys are not empty during node initialization, because this is unrecoverable error that can be detected even before node starts.